packages/SimPKPD/src/pk-pd-tools.ts

```diff
@@ -0,0 +1,165 @@
+function getSimFn(compartments: string): any {
+  switch (compartments) {
+    case '1 compartment PK':
+      return simulateOneCompartmentPK;
+
+    case '2 compartment PK':
+      return simulateTwoCompartmentPK;
+
+    default:
+      throw new Error('Unsupported PK/PD model.');
+  }
+}
+
+/** Simulate PK-PD on the one dosage interval */
+async function oneIntervalSimulation(compartments: string,
+  initial: number, final: number, step: number,
+  _depotInitial: number, _centrInitial: number, _periInitial: number, _effInitial: number, 
+  _KAVal: number, _CLVal: number, _V2Val: number, _QVal: number, _V3Val: number, _KinVal: number, _KoutVal: number, _EC50Val: number): Promise<DG.DataFrame>
+{
+  // Specify modelling function
+  const simFn = getSimFn(compartments);
+
+  // Perform wasm-computations
+  const res = await simFn(initial, final, step,
+    _depotInitial, _centrInitial, _periInitial, _effInitial, 
+    _KAVal, _CLVal, _V2Val, _QVal, _V3Val, _KinVal, _KoutVal, _EC50Val) as DG.DataFrame;
+
+  // Compute concentrations
+  const conc = getConc(res.getCol(CENTR_T), res.getCol(PERI_T), _V2Val, _V3Val);
+
+  // Add concentrations to the results
+  res.columns.add(conc.C2);
+  res.columns.add(conc.C3);
+
+  return res;
+}
+
+/** Simulate PK-PD */
+export async function simPKPD(compartments: string,
+  dose: number, dosesCount: number, doseInterval: number,
+  _KAVal: number, _CLVal: number, _V2Val: number, _QVal: number, _V3Val: number, _KinVal: number, _KoutVal: number, _EC50Val: number): Promise<DG.DataFrame>
+{
+  // Initial values
+  let _depotInitial = dose;
+  let _centrInitial = 0;
+  let _periInitial = 0;
+  let _effInitial = 1;
+
+  // Independent variable range
+  let initial = 0;
+  let final = doseInterval;
+  const step = doseInterval / POINTS_COUNTER;
+
+  // Index of the last row
+  let lastIdx;
+
+  // Get simulation on the first interval
+  const res = await oneIntervalSimulation(compartments, 
+    initial, final, step,
+    _depotInitial, _centrInitial, _periInitial, _effInitial, 
+    _KAVal, _CLVal, _V2Val, _QVal, _V3Val, _KinVal, _KoutVal, _EC50Val);
+
+  // Get simulations on the next intervals
+  for (let i = 1; i < dosesCount; ++i) {
+    // Update independent variable range
+    initial = final;
+    final += doseInterval;
+
+    lastIdx = res.rowCount - 1;
+    
+    // Update initial values
+    _depotInitial = res.get(DEPOT_T, lastIdx) + dose;    
+    _centrInitial = res.get(CENTR_T, lastIdx);
+    _periInitial = res.get(PERI_T, lastIdx);
+    _effInitial = res.get(EFF_T, lastIdx);
+
+    // Compute simulation & update results
+    res.append(
+      await oneIntervalSimulation(compartments, 
+        initial, final, step,
+        _depotInitial, _centrInitial, _periInitial, _effInitial, 
+        _KAVal, _CLVal, _V2Val, _QVal, _V3Val, _KinVal, _KoutVal, _EC50Val),
+      true
+    );
+  }
+
+  // Rename columns
+  const timeCol = res.col(T_COL);
```

Comment:
I think it's would be much shorter and a lot more clear to just have 5 lines instead of 15 like that:

```
res.col(T_COL)!.name = TIME;
res.col(EFF_T)!.name = EFFECT;
```

Reply:
Or simply use the same constant; or, define mapping in the dictionary

Reply:
Done